gamla/graph.py

```diff
@@ -8,29 +8,63 @@
 from gamla import functional, functional_generic
 
 
+@toolz.curry
+def _is_better_node(node: Any, seen: Dict, key: Callable):
+    old_node = seen[key(node)]
+    return node > old_node
+
+
+@toolz.curry
+def _save_to_dict(element: Any, dictionary: Dict, func: Callable):
```

Comment:
when currying, place the variables in the order you expect to use them, so element would be last in this case

Reply:
Ok!

Reply:
By doing so I get an error, 
`TypeError: _save_to_dict() got multiple values for argument 'dictionary'`

Reply:
try removing the keywords, it doesn't mix well with currying
so just call the variables by the order of the signature (which should be `dictionary, func, element`)